packages/netlify-cms-core/src/actions/entries.ts

```diff
@@ -815,3 +830,55 @@ export function deleteEntry(collection: Collection, slug: string) {
       });
   };
 }
+
+const getPathError = (
+  path: string | undefined,
+  key: string,
+  t: (key: string, args: Record<string, unknown>) => string,
+) => {
+  return {
+    error: {
+      type: ValidationErrorTypes.CUSTOM,
+      message: t(`editor.editorControlPane.widget.${key}`, {
+        path,
+      }),
+    },
+  };
+};
+
+export function validateMetaField(
+  state: State,
+  collection: Collection,
+  field: EntryField,
+  value: string | undefined,
+  t: (key: string, args: Record<string, unknown>) => string,
+) {
+  if (field.get('meta') && field.get('name') === 'path') {
+    if (!value) {
+      return getPathError(value, 'invalidPath', t);
+    }
+    const sanitizedPath = (value as string)
+      .split('/')
+      .map(getProcessSegment(state.config.get('slug')))
+      .join('/');
+
+    if (value !== sanitizedPath) {
+      return getPathError(value, 'invalidPath', t);
+    }
+
+    const customPath =
+      selectCustomPath(collection, fromJS({ entry: { meta: { path: value } } })) || '';
+
+    const existingEntry =
+      customPath && selectEntryByPath(state.entries, collection.get('name'), customPath);
+
+    const existingEntryPath = existingEntry && existingEntry.get('path');
+    const entryDraft = state.entryDraft;
+    const draftPath = entryDraft && entryDraft.getIn(['entry', 'path']);
+
+    if (existingEntryPath && existingEntryPath !== draftPath) {
```

Comment:
`existingEntryPath !== draftPath` since we don't want to fail the validation when editing an existing entry (`selectEntryByPath` search all entries).